[ENG-7305] Enable to Withdraw Preprint Again After the preprint has been unwithdrawn

bodintsov opened the pull request:
## Purpose

Enable to Withdraw Preprint Again After the preprint has been unwithdrawn

## Changes

TBD

## QA Notes

TBD

## Documentation

TBD

## Side Effects

TBD

## Ticket
(https://openscience.atlassian.net/browse/ENG-7305)

brianjgeiger commented on admin/preprints/views.py:
```diff
@@ -591,6 +591,7 @@ def post(self, request, *args, **kwargs):
 
         from osf.utils.migrations import disable_auto_now_fields
         with disable_auto_now_fields():
+            preprint.requests.all().delete()
```
So is there not a middle ground between not deleting any requests and deleting all the requests?

brianjgeiger reviewed (changes requested):
One question

bodintsov commented:
> Two things:
> 
> 1. The comment in the code to modify the test to withdraw the preprint again to make sure that works
> 2. Could you make a second test where the admins approve the withdrawal request, then unwithdraw it, then withdraw it again?

Updated test. Updated implementation of the test covers the second point. Workflow of the testcase: 1. approve preprint withdrawal request. 2. unwithdraw preprint. 3. Create a new withdrawal request. 4 Approve withdrawal request. I think there is no need to add a second test as this one covers it and it will be 2 identical tests. Maybe I missed something?
